can/can.cpp

```diff
@@ -0,0 +1,255 @@
+#ifndef CONFIG_DISABLE_MODULE_CAN
+
+#include "can.hpp"
+
+namespace halex {
+
+Can::Can(CAN_HandleTypeDef *hcan) : hcan(hcan) {
+
+}
+
+CAN_StatusType Can::init() {
+    if (HAL_CAN_Init(hcan) != HAL_OK) {
+        return CAN_StatusType::CAN_Fail_Init;
+    }
+    return static_cast<CAN_StatusType>(HAL_CAN_Start(hcan));
+}
```

Comment:
```suggestion
CAN_StatusType Can::setup() {
    if (HAL_CAN_Init(hcan) != HAL_OK) {
        return CAN_StatusType::CAN_Fail_Init;
    }
    return static_cast<CAN_StatusType>(HAL_CAN_Start(hcan));
}
```

[WANT] init と start の両方を行うので、init や start 以外の名前を使いたいです。わたしであれば setup にします

Reply:
6a6828c9f48125adb2a0f35838d9bda6e6626f7e で対応